marklogic-data-hub-central/src/test/java/com/marklogic/hub/central/controllers/EntitySearchControllerTest.java

```diff
@@ -108,9 +108,17 @@ void testCRUDOnSavedQuery() throws Exception {
         verifyRequestIsForbidden(MockMvcRequestBuilders.delete(SAVED_QUERIES_PATH + "/query").param("id", "some-id"));
     }
 
+    // This test is done at the Java layer as multiple scenarios cannot be tested in ml-unit-tests due to the optic calls occurring in the same request.
+    // There appears to be a bug with Optic caching values in the same request that is being investigated.
     @Test
     void testGetDocumentByURI() throws Exception {
         ReferenceModelProject project = installOnlyReferenceModelEntities(true);
+        project.setCustomerDocumentMetadata(new DocumentMetadataHandle()
```

Comment:
It's difficult to read this test and see that the 3 scenarios called out in the story - no prov records, 1 prov record, and 2 prov records - are being tested. 

Also, I'd much rather see this tests in core DHF as opposed to HC. In HC, a single smoke test will do the trick. 

So move the tests to core DHF, write the test against the EntitySearchService - call it GetRecordTest, since you're testing the getRecord endpoint - and I recommend a separate test method for each of the 3 scenarios. The steps don't matter. So for the first one, you can just ingest a record. For the second one, ingest it and map it. For the 3rd, ingest it, map it, and maybe use a custom step (or 2 custom steps instead of map). All values in each row in the history object need to be verified.

Reply:
There is a getRecordHistoryTest.sjs for that covers all the scenarios. getRecord.sjs test is failing as multiple scenarios cannot be tested in ml-unit-tests due to the optic calls occurring in the same request. There appears to be a bug with Optic caching values in the same request that is being investigated.